modules/Kconfig.st

```diff
@@ -4,3 +4,126 @@
 
 config HAS_STLIB
     bool
+
+config HAS_STMEMSC
+	bool
+	imply NEWLIB_LIBC
```

Comment:
I'm curious why not using select here?

Reply:
> I'm curious why not using select here?

@erwango 
Because NEWLIB_LIBC is a visible symbol, i.e. with a prompt.

If you see the kernel kbuild documentation
(https://www.kernel.org/doc/Documentation/kbuild/kconfig-language.txt)

it clearly states that it is not recommended to use select in this case:

Note:
	select should be used with care. select will force
	a symbol to a value without visiting the dependencies.
	By abusing select you are able to select a symbol FOO even
	if FOO depends on BAR that is not set.
	In general use select only for non-visible symbols
	(no prompts anywhere) and for symbols with no dependencies.
	That will limit the usefulness but on the other hand avoid
	the illegal configurations all over.

Reply:
ok, clear. Thanks!

Reply:
usage of 'imply' was a suggestion from @galak